sycl/doc/extensions/proposed/sycl_ext_oneapi_async_memory_alloc.asciidoc

```diff
@@ -0,0 +1,838 @@
+those modifications reflected as it conforms to the SYCL common reference
+semantics.
+
+[source,c++]
+----
+class context {
+
+  memory_pool context::ext_oneapi_get_default_memory_pool(usm::alloc kind)
+    const;
+
+  memory_pool context::ext_oneapi_get_default_memory_pool(const device &dev,
+    usm::alloc kind) const;
+
+}; // context
+----
+
+[source, c++]
+----
+memory_pool context::ext_oneapi_get_default_memory_pool(usm::alloc kind) const;
+----
+
+_Returns_: The default memory pool associated with the context for allocating
+with the allocation kind `kind`.
+
+_Throws_: An exception with the `errc::invalid` error code if `kind` is
+`usm::alloc::device`, `usm::alloc::shared` or `usm::alloc::unknown`.
+
+[source, c++]
+----
+memory_pool context::ext_oneapi_get_default_memory_pool(const device &dev,
+  usm::alloc kind) const;
+----
+
+_Returns_: The default memory pool associated with the context and `dev` for
+allocating with the allocations of kind `kind`.
+
+_Throws_: An exception with the `errc::invalid` error code if `dev` is not one
+of the devices associated with the context. An exception with the
+`errc::invalid` error code if `kind` is `usm::alloc::host` or
+`usm::alloc::unknown`.
+
+
+=== Asynchronous malloc & free
+
+This extension introduces a series of new enqueue functions for enqueueing
+asynchronous malloc and free commands which operate with the memory pools also
+introduced in this extension.
+
+All enqueue functions introduced have overloads which take a SYCL `queue` and a
+SYCL `handler`. None of enqueue functions return a SYCL `event` directly, as
+this extension is in line with the
+link:../experimental/sycl_ext_oneapi_enqueue_functions.asciidoc[
+  sycl_ext_oneapi_enqueue_functions] extension, so events are returned when
+calling `submit_with_event` and the `handler` overloads of these enqueue
+functions.
+
+[source,c++]
+----
+namespace ext::oneapi::experimental {
+
+void *async_malloc(const queue &que, usm::alloc kind, size_t size);
+
+void *async_malloc(handler &cgh, usm::alloc kind, size_t size);
+
+void *async_malloc_from_pool(const queue &que, size_t size,
+  const memory_pool &pool);
+
+void *async_malloc_from_pool(handler &cgh, size_t size,
+  const memory_pool &pool);
+
+void async_free(const queue &que, void *ptr);
+
+void async_free(handler &cgh, void *ptr);
+
+}  // ext::oneapi::experimental
+----
+
+[source, c++]
+----
+void *async_malloc(const queue &que, usm::alloc kind, size_t size);
+
+void *async_malloc(handler &cgh, usm::alloc kind, size_t size);
+
+void *async_malloc_from_pool(const queue &que, size_t size,
+  const smemory_pool &pool);
+
+void *async_malloc_from_pool(handler &cgh, size_t size,
+  const memory_pool &pool);
+----
+
+_Effects_: Enqueues a command to `q` or the SYCL queue associated with `h` which
+will asynchronously allocate memory of size `size` in bytes, allocating from the
+memory pool `pool` if provided, otherwise allocation from the default memory
```

Comment:
```suggestion
memory pool `pool` if provided, otherwise allocating from the default memory
```

Reply:
Fixed in https://github.com/intel/llvm/pull/14800/commits/d154a67861e46477ba07b415b845348f880ce8a4.